src/main/java/duke/interaction/Parser.java

```diff
@@ -0,0 +1,68 @@
+package duke.interaction;
+
+import duke.command.Command;
+import duke.command.AddCommand;
+import duke.command.ListCommand;
+import duke.command.ExitCommand;
+import duke.command.DoneCommand;
+import duke.command.DateCommand;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.HelpCommand;
+import duke.task.Task;
+import duke.util.DukeException;
+
+import java.util.Scanner;
+
+/**
+ * Handles the applications parsing behaviour,
+ * deciding what action to take depending on the user input/command.
+ *
+ * @author  Hardy Shein
+ * @version 0.1
+ */
+public class Parser {
+
+    /**
+     * Analyses a full command string and converts
+     * to an executable Command object.
+     * @param fullCommand read from user input.
+     * @return Command object parsed from a full command string
+     */
+    public static Command parse(String fullCommand) throws DukeException.InvalidCommand {
+        Scanner in = new Scanner(fullCommand);
+        String input = in.next();
+        try {
+            switch (input.toLowerCase()) {
+            case "list":
+                return new ListCommand();
+            case "bye":
+                return new ExitCommand();
+            case "done":
+                return new DoneCommand((in.hasNextInt() ? in.nextInt() - 1 : -1));
```

Comment:
Maybe have methods for the corresponding arguments in creating the different types of commands, it might help in code readability.